### PR TITLE
SDK Architecture: Marmade js

### PR DESCRIPTION
**Pull request by anxolin**
Attempt to replace Plant UML for our documentation of the Architecture.

This is the proposal:
https://github.com/cowprotocol/cow-sdk/blob/marmade-js/docs/architecture.md
image

> 🚨 The graphs where up to date when I did them, now it will be outdated specially the part of the metadata API now being part of the SDK. I will deal with this in another PR, and probably @Sasha will want to reiterate and update a few things from the diagrams. I just want to have this base merged asap so we can re-iterate

## More Context on this PR

The advantage of Marmade JS is:
- It's more modern
- Better documentation (my eyes cry when  i try to read https://plantuml.com/, but this is niceeee https://mermaid.js.org/)
- A lot of nice diagrams we can use! (Gantt, Pie, Flow, State, Sequence, ...)

But the most important for me:
- Works out of the box with Github
- Easy to include in the markdown, so you can add also some text

If this is not enough, it also works with Notion and many other tools, so if we learn Marmade-js we can use it there too :) 

## Scope

- Creates a new architecture MARKDOWN file in the doc, and link it from the main README
- Deletes the old PLANT UML
- Creates some high level diagrams (architecture, model)
- Create some more detail ones: order model, order book API, subgraph API, metadata API

I decided to split the APIs in sections, and to not define most of the model in the API to keep it simpler. 

## Not in scope
I didn't try to be exhaustive, so there must be a few differences. Sometimes I would remove some information just to make it more readable. 

The main idea, is to get an overview idea of all the APIS, models, etc so it's easier to use the SDK

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+        fill: #eeff82
+    }
+
+     
+     .important > rect {
+        stroke-width: 3;
+        stoke: #eeff82;
+        fill: #eeff82
+    }
+</style>
```
Comment (alfetopito) [inline docs/architecture.md:50]: This does not seem to render, at least not in the PR

image

File: docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+
+class OrderParameters {
+  + sellAmount: TokenAmount;
+  + buyAmount: TokenAmount;
+  + validTo: number;
+  + feeAmount: TokenAmount;
+  + kind: OrderKind;
+  + partiallyFillable: boolean;
+}
```
[the remark this answers is not in the document]
Comment (anxolin) [reply docs/architecture.md:63]: I think so, although auto-generate will also be a bit exhaustive. For some things in UML in think is nice to omit details

File: docs/architecture.md (hunk 1 above)
Comment (anxolin) [reply docs/architecture.md:50]: Removed

File: docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+SDK --> OrderBookApi
+SDK --> SubgraphApi
+SDK --> MetadataApi
+```
+
+The SDK has 3 main APIs
+
+- **Order Book API**: Allows to get the open orders, historic orders, post new orders, etc.
+- **Subgraph API**: Provides access to On-chain data indexed by The Graph
+- **Metadata API**: Allows to encode/decode meta-data to be attached in orders
```
Comment (alfetopito) [inline docs/architecture.md:20]: This has actually been removed/moved to @cowprotocol/app-data

**Review by alfetopito — APPROVED** [review]
Approved, with the caveat that Metadata api is no longer in the sdk

File: docs/architecture.md (hunk 3 above)
Comment (anxolin) [reply docs/architecture.md:20]: Yep, I know :) 

I consider deleting, but I think we should move the docs to that project too. And I don't want to step too much in the Sasha's part. I just wanted to try the PLANTUML and make a proposal with Marmade UML. 

Let's merge and then we can continue to move things around and do the finer adjustments to it

**Pull request merged**